sql/updates/latest-dev.sql

```diff
@@ -0,0 +1,10 @@
+CREATE TABLE IF NOT EXISTS _timescaledb_catalog.continuous_aggs_bucket_function(
+  mat_hypertable_id integer PRIMARY KEY REFERENCES _timescaledb_catalog.hypertable (id) ON DELETE CASCADE,
```

Comment:
Just wanted to add that I disagree on this one:

> A constant origin does not make sense for caggs

Firstly, `origin` will be useful when people will migrate from time_bucket to time_bucket_ng() since these functions have different values of default `origin`.

Monday, Jan 3 was chosen for time_bucket() because it makes more sense for weekly buckets, and Jan 1 for time_bucket_ng() because Jan 3 doesn't make sense for monthly buckets. We didn't choose a particular strategy of promoting time_bukcet_ng() from the experiment schema yet, but one of the strategies is to keep both functions for several releases and let users do the migration manually. They will be unable to do it without specifying `origin`.

Secondly, in some countries Monday is considered to be the beginning of the week, while in others it's Sunday. Specifying `origin` will allow users to choose what is more convenient for them.

Finally, there is little reason NOT to support this feature. It's easy to implement and it costs nothing to us.

Reply:
Maybe instead of having an `experimental` field the relation itself could be named experimental-something? This would show that this metadata table is not part of the stable schema, instead of having per record state.

Reply:
@mfundul Experimental column here is a compact way to store the schema of the bucketing function. This information is not used at the moment for anything, but will be needed when we will be switching from time_bucket() to time_bucket_ng().

In other words:

time_bukcet() = [experimental = false, name = "time_bucket"]
time_bucket_ng() = [experimental = true, name = "time_bucket_ng"]

During the promotion of time_bucket_ng() we may also have a situation when:

time_bucket_ng() = [experimental = false, name = "time_bucket_ng"]

Reply:
This is not quite obvious from the code though. I will add extra comments to the ContinuousAggsBucketFunction structure 👍

Reply:
There is a good chance that when the feature stabilizes (after the time_bucket_ng() promotion and whatnot) we will end up without `experimental` and `name` columns. So I wouldn't invest too much time into discussing the exact schema until then.

Reply:
I'm marking this long discussion as resolved, see the summary https://github.com/timescale/timescaledb/pull/3753#issuecomment-991073467